packages/0x.js/src/0x.ts

```diff
@@ -238,19 +244,19 @@ export class ZeroEx {
      * @param   orderHash       Hex encoded orderHash to sign.
      * @param   signerAddress   The hex encoded Ethereum address you wish to sign it with. This address
      *          must be available via the Provider supplied to 0x.js.
-     * @param   MessagePrefixOpts  Options regarding the desired prefix and whether to add it before calling `eth_sign`
-     * @return  An object containing the Elliptic curve signature parameters generated by signing the orderHash.
+     * @param   SignerProviderType  The type of Signer Provider which implements `eth_sign`. E.g Metamask, Ledger, Trezor or EthSign.
```

Comment:
This is a bit confusing, how can `EthSign` be a signer provider type? Maybe we just call it `Other`? There are definitively other signers out there. We can also add a note to this `@param` that `Other` assumes the prefix is added by the signer, and uses the prefix specified in the Ethereum wiki docs.

Reply:
I'm also not a fan of the naming `SignerProviderType`, why not just `BackingSigner`? Or `SignerType`?

Reply:
Any implementation which implements the EthSign as per the spec is an EthSign provider (regardless of where the key lives and signing process happens). Anything else is out of spec to EthSign and should be separately called out. Eventually we would only have two options, Spec Compliant (EthSign, as in eth_sign) and Trezor. 

There are definitely other signers out there, but they would either be spec compliant (EthSign) or we would flag them as their own SignerProviderType to implement the workarounds.

`BackingSigner` sounds like it should have available options for `PrivateKey` and `Mnemonic` which doesn't make much sense here.

Reply:
I think we are on the same page but speaking past each other a bit. I'm proposing we do:

```
* @param   SignerType  The type of signer that will execute `eth_sign`. E.g Metamask, Trezor or Other. (note: `Other` will act in a `eth_sign` spec-compliant manner)
```

This makes it clear that `Metamask` and `Trezor` are the anomalies here and everything else can just pass in `Other` here.

Re: Knowing which are r,s,v/v,r,s, at one point Parity & Geth differed here. This might have been fixed at some point, but we kept it to remain backward-compatible. We could probably remove this at some point too.